toolchain/sem_ir/function.h

```diff
@@ -19,7 +19,12 @@ namespace Carbon::SemIR {
 // Function-specific fields.
 struct FunctionFields {
   // Kinds of special functions.
-  enum class SpecialFunctionKind : uint8_t { None, Builtin, Thunk };
+  enum class SpecialFunctionKind : uint8_t {
+    None,
+    Builtin,
+    Thunk,
+    CppThunkRequired,
```

Comment:
I found the `Required` here a bit surprising at first, but I think I understand -- you're saying that this function isn't itself a thunk, but it has a thunk and so shouldn't be called directly. Maybe `HasCppThunk` would be a bit clearer than `CppThunkRequired`. Would that work for you?

Reply:
Done.